lib/request_handler/body_parser.rb

```diff
@@ -19,12 +30,17 @@ def flattened_request_body
       body = request_body.fetch('data') do
         raise ExternalArgumentError, body: 'must contain data'
       end
-      body.merge!(body.delete('attributes') { {} })
-      relationships = flatten_relationship_resource_linkages(body.delete('relationships') { {} })
-      body.merge!(relationships)
+      flatten_resource!(body)
```

Comment:
I think we could always return

```ruby
[flatten_resource!(body), *parsed_included]
```

and do the `included_schemas?` check only in the `parse_included`